sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -6,28 +6,39 @@
 from pyro.infer.mcmc.api import MCMC
 from sbi.mcmc import Slice
 from torch import multiprocessing as mp
-
-from pyknos.nflows import flows
 from sbi.mcmc import SliceSampler
+import sbi.inference
 
 
-class FlowPosterior(flows.Flow):
+class Posterior:
     def __init__(
         self,
-        transform,
-        distribution,
+        algorithm,
+        neural_net,
         prior,
         context,
-        embedding=None,
-        train_with_mcmc=False,
+        train_with_mcmc=True,
         mcmc_method="slice-np",
     ):
+        """
+        Args:
+            algorithm: string, 'snpe', 'snl', 'sre'
+            neural_net: depends on algorithm: classifier for sre, density estimator for snpe and snl
+            prior: prior dist
+            context: Tensor or None, conditioning variables. If a Tensor, it must have the same
+                number or rows as the inputs. If None, the context is ignored.
+            train_with_mcmc: bool. Sample rejection or MCMC?
 
-        super().__init__(transform, distribution, embedding)
+        Returns:
+            A Tensor of shape [input_size], the log probability of the inputs given the context.
+        """
+
+        self.neural_net = neural_net
```

Comment:
Is there a reason behind sometimes using an underscore, and sometime not when adding to `self.`?